authorizer.py

```diff
@@ -85,6 +87,9 @@ def _VerifyServingSession(self, institution, session):
     if (serving_session.login_type == "schedule" and
         self.handler.request.path == "/spots_available"):
```

Comment:
Actually I take the above comment back. I think it would be much better if each page is responsible for deciding when it can be shown. I think we could completely delete the _VerifyServingSession method. Also delete where it's used on lines 75 and 76 above. Then each page would have to check both student access and serving session as follows:

```python
if not auth.HasStudentAccess():
  auth.Redirect()
  return
serving_session = models.ServingSession.FetchEntity(institution)
# decide which serving sessions are OK for this page.
if serving_session != "schedule":
  auth.Redirect()
  return
```